<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GcRailRobotClass</class>
 <widget class="QMainWindow" name="GcRailRobotClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1029</width>
    <height>665</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GcRailRobot</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_8">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Device IP:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEdit_ip"/>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="conn_btn">
            <property name="text">
             <string>connect</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Port:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_port"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="common_widget" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>9</number>
             </property>
             <item>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Mode Set</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="4" colspan="2">
                <widget class="QPushButton" name="send_mode_btn">
                 <property name="text">
                  <string>send_mode</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="comboBox_mode"/>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Speed{/r/min}</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="5">
                <widget class="QPushButton" name="send_speed_btn">
                 <property name="text">
                  <string>send_speed</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1" colspan="4">
                <widget class="QLineEdit" name="lineEdit"/>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout">
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>Write Data{common data}</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QTextEdit" name="textEdit"/>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout">
                 <item>
                  <spacer name="horizontalSpacer_2">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="QPushButton" name="pushButton_2">
                   <property name="text">
                    <string>send</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_3">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Recv Data from Device</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_recv"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Debug output:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="textEdit_debug">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>70</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1029</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="GcRailRobot.qrc"/>
 </resources>
 <connections/>
</ui>
